tesseract_common/src/manipulator_info.cpp

```diff
@@ -81,152 +83,30 @@ const Eigen::Isometry3d& ToolCenterPoint::getTransform() const
-          Eigen::AngleAxisd yawAngle(y, Eigen::Vector3d::UnitZ());
-
-          Eigen::Quaterniond rpy = yawAngle * pitchAngle * rollAngle;
-
-          local_tcp.linear() = rpy.toRotationMatrix();
-        }
-      }
-      else
-      {
-        status = QueryStringAttribute(tcp_element, "wxyz", wxyz_string);
-        if (status != tinyxml2::XML_NO_ATTRIBUTE && status != tinyxml2::XML_SUCCESS)
-          throw std::runtime_error("ManipulatorInfo: Error parsing TCP attribute wxyz.");
-
-        if (status != tinyxml2::XML_NO_ATTRIBUTE)
-        {
-          std::vector<std::string> tokens;
-          boost::split(tokens, wxyz_string, boost::is_any_of(" "), boost::token_compress_on);
-          if (tokens.size() != 4 || !isNumeric(tokens))
-            throw std::runtime_error("ManipulatorInfo: Error parsing TCP attribute wxyz string.");
-
-          double qw{ 0 }, qx{ 0 }, qy{ 0 }, qz{ 0 };
-          // No need to check return values because the tokens are verified above
-          toNumeric<double>(tokens[0], qw);
-          toNumeric<double>(tokens[1], qx);
-          toNumeric<double>(tokens[2], qy);
-          toNumeric<double>(tokens[3], qz);
-
-          Eigen::Quaterniond q(qw, qx, qy, qz);
-          q.normalize();
-
-          local_tcp.linear() = q.toRotationMatrix();
-        }
-      }
-      tcp = local_tcp;
-    }
-    else
-    {
-      std::string tcp_name;
-      tinyxml2::XMLError status = QueryStringAttribute(tcp_element, "name", tcp_name);
-      if (status != tinyxml2::XML_SUCCESS)
-        throw std::runtime_error("ManipulatorInfo: Error parsing TCP attribute name.");
-
-      tcp = tcp_name;
-    }
-
-    if (tcp_element->Attribute("external") != nullptr)
-    {
-      bool external = false;
-      tinyxml2::XMLError status = tcp_element->QueryBoolAttribute("external", &external);
-
-      if (status != tinyxml2::XML_SUCCESS)
-        throw std::runtime_error("ManipulatorInfo: Error parsing TCP attribute external.");
-
-      std::string external_frame;
-      if (tcp_element->Attribute("external_frame") != nullptr)
-      {
-        tinyxml2::XMLError status = QueryStringAttribute(tcp_element, "external_frame", external_frame);
-        if (status != tinyxml2::XML_SUCCESS)
-          throw std::runtime_error("ManipulatorInfo: Error parsing TCP attribute external_frame.");
-      }
+  bool ret_val = true;
+  ret_val &= (type_ == rhs.type_);
+  ret_val &= (name_ == rhs.name_);
+  ret_val &= (transform_.isApprox(rhs.transform_, 1e-5));
+  ret_val &= (external_ == rhs.external_);
+  ret_val &= (external_frame_ == rhs.external_frame_);
+  return ret_val;
+}
+bool ToolCenterPoint::operator!=(const ToolCenterPoint& rhs) const { return !operator==(rhs); }
 
-      tcp.setExternal(external, external_frame);
-    }
-  }
+template <class Archive>
+void ToolCenterPoint::serialize(Archive& ar, const unsigned int /*version*/)  // NOLINT
+{
+  ar& boost::serialization::make_nvp("type", type_);
```

Comment:
I notice about you used `BOOST_SERIALIZATION_NVP`. It looks like this probably doesn't matter, but is there any reason that we should be using the macro instead of this?

Reply:
I think the main difference is that you can specify the name in the NVP using the non-macro method. The macro uses the name of the variable as the name in the NVP. Not sure if the manually specified name is worth keeping or not

Reply:
Yea it was more for esthetics. I did not what the name of the variable in the xml to have underscore though it may be good to keep them the same for traceability. What do you all thing?